src/docfx/build/toc/TableOfContentsMap.cs

```diff
@@ -112,9 +97,71 @@ private static (int subDirectoryCount, int parentDirectoryCount) GetDirectoryCou
                         break;
                 }
             }
-
-            subDirectoryCount = relativePathParts.Length - parentDirectoryCount - 1;
+            subDirectoryCount = relativePathParts.Count() - parentDirectoryCount;
             return (subDirectoryCount, parentDirectoryCount);
         }
+
+        private sealed class TocCandidate
+        {
+            public int SubDirectoryCount { get; }
+
+            public int ParentDirectoryCount { get; }
+
+            // save the calculated distance during comparison to avoid repeadly calculation
+            private readonly Lazy<int> _levenshteinDistance;
+
+            public int LevenshteinDistance => _levenshteinDistance.Value;
+
+            public Document Toc { get; }
+
+            public Document TargetFile { get; }
+
+            public TocCandidate(int subDirectoryCount, int parentDirectoryCount, Document toc, Document targetFile)
+            {
+                SubDirectoryCount = subDirectoryCount;
+                ParentDirectoryCount = parentDirectoryCount;
+                Toc = toc;
+                TargetFile = targetFile;
+                _levenshteinDistance = new Lazy<int>(() => GetLevenshteinDistanceToFile());
+            }
+
+            private int GetLevenshteinDistanceToFile()
+            {
+                var fileNames = Path.GetFileNameWithoutExtension(TargetFile.SitePath).Split(new[] { '-', '#', '_' })
```

Comment:
>Path.GetFileNameWithoutExtension(TargetFile.SitePath).Split(new[] { '-', '#', '_' }) [](start = 32, length = 84)

make this as a help method, once you want to add new splitter,  u don't need change two places